tests/playbooks/tests_route_device.yml

```diff
@@ -0,0 +1,188 @@
+            specified
+          import_role:
+            name: linux-system-roles.network
+          vars:
+            network_connections:
+              - name: "{{ interface0 }}"
+                interface_name: "{{ interface0 }}"
+                state: up
+                type: ethernet
+                autoconnect: false
+                ip:
+                  address:
+                    - 198.51.100.3/24
+                    - 2001:db8::2/32
+                  route:
+                    - network: 198.51.10.64
+                      prefix: 26
+                      gateway: 198.51.100.6
+                      metric: 4
+                    - network: 2001:db6::4
+                      prefix: 128
+                      gateway: 2001:db8::1
+                      metric: 2
+              - name: "{{ interface1 }}"
+                interface_name: "{{ interface1 }}"
+                state: up
+                type: ethernet
+                autoconnect: false
+                ip:
+                  address:
+                    - 198.51.100.6/24
+                    - 2001:db8::4/32
+                  route:
+                    - network: 198.51.12.128
+                      prefix: 26
+                      gateway: 198.51.100.1
+                      metric: 2
+        - name: Get the IPv4 routes from the route table main
+          command: ip -4 route
+          register: route_table_main_ipv4
+          changed_when: false
+
+        - name: Assert that the route table main contains the specified IPv4
+            routes
+          assert:
+            that:
+              # When running with nm provider, the route will be configured
+              # with `proto static`
+              # In RHEL-6.10 managed host, the attribute in the route is not
+              # equally spaced
+              - route_table_main_ipv4.stdout is search("198.51.10.64/26 via
+                198.51.100.6 dev ethtest0\s+(proto static )?metric 4")
+              - route_table_main_ipv4.stdout is search("198.51.12.128/26 via
+                198.51.100.1 dev ethtest1\s+(proto static )?metric 2")
+            msg: "the route table main does not contain the specified IPv4
+              route"
+
+        - name: Get the IPv6 routes from the route table main
+          command: ip -6 route
+          register: route_table_main_ipv6
+          changed_when: false
+
+        - name: Assert that the route table main contains the specified IPv6
+            routes
+          assert:
+            that:
+              - route_table_main_ipv6.stdout is search("2001:db6::4 via
+                2001:db8::1 dev ethtest0\s+(proto static )?metric 2")
+            msg: "the route table main does not contain the specified IPv6
+              route"
+        - name: Get the interface1 MAC address
+          command: cat /sys/class/net/{{ interface1 }}/address
+          register: interface1_mac
+          ignore_errors: true  # noqa ignore-errors
```

Comment:
What's the reason to have ignore_errors:true here?

Reply:
I was thinking that the interface1 is possible not created and configured properly. So the  directory `/sys/class/net/{{ interface1 }}/` may not exist, but after the second thought, drop `ignore_errors: true  # noqa ignore-errors`  is better, so that error will be raised when  the  directory `/sys/class/net/{{ interface1 }}/` is not existed.